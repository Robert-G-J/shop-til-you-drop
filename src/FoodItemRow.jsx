import React from "react";

const FoodItemRow = () => (
  <div className="food-item-row">
    <div className="food-item-row__item" />
    <div className="food-item-row__price" />
    <button className="button-increment">+</button>
    <button className="button-decrement">-</button>
    <div className="food-item-row__quantity" />
    <div className="food-item-row__subtotal" />
  </div>
);

export default FoodItemRow;
