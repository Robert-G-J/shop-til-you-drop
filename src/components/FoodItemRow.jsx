import React from "react";
import PropTypes from "prop-types";

const FoodItemRow = props => (
  <div className="food-item-row">
    <div className="food-item-row__item">{props.type}</div>
    <div className="food-item-row__price">Â£{props.price}</div>
    <div className="food-item-row__unit">per {props.unit}</div>
    <button className="button-increment">+</button>
    <button className="button-decrement">-</button>
    <div className="food-item-row__quantity" />
    <div className="food-item-row__subtotal" />
  </div>
);

FoodItemRow.propTypes = {
  props: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
      unit: PropTypes.string.isRequired
    })
  )
};

export default FoodItemRow;
