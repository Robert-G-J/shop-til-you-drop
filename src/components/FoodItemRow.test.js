/* eslint-disable */

import React from "react";
import FoodItemRow from "./FoodItemRow";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("The shopping cart app", () => {
  let component;
  const props = {
    id: 1,
    type: "peas",
    price: "2.10",
    unit: "bag"
  };
  beforeEach(() => {
    component = shallow(<FoodItemRow {...props} />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<FoodItemRow {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("doesn't crash", () => {
    expect(component.exists()).toEqual(true);
  });

  describe("has buttons", () => {
    it("to increment", () => {
      expect(component.find(".button-increment").length).toEqual(1);
    });

    it("to decrement", () => {
      expect(component.find(".button-decrement").length).toEqual(1);
    });
  });
});
