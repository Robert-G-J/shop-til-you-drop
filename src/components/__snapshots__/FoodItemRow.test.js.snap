// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The shopping cart app renders correctly 1`] = `
<div
  className="table-row"
>
  <div
    className="table-row__product table-row__text"
  >
    Peas
  </div>
  <div
    className="table-row__price table-row__text"
  >
    Â£
    2.10
  </div>
  <div
    className="table-row__unit table-row__text"
  >
    per 
    bag
  </div>
  <div
    className="table-row__text"
  >
    <button
      className="button-increment"
    >
      +
    </button>
  </div>
  <div
    className="table-row__text"
  >
    <button
      className="button-decrement"
    >
      -
    </button>
  </div>
  <div
    className="table-row__quantity table-row__text"
  >
    0
  </div>
  <div
    className="table-row__subtotal table-row__text"
  >
    0
  </div>
</div>
`;
